src/views/Profile/AuthorsList/AuthorsList.tsx

```diff
@@ -1,113 +1,95 @@
-import React from 'react';
-import { Paper, Table, TableContainer, Box, Button } from '@material-ui/core';
-import { IAdminLabExpert } from 'models/adminLab/types';
+import React, { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  Paper,
+  Table,
+  TableContainer,
+  Box,
+  Button,
+  Typography,
+} from '@material-ui/core';
+
 import AuthorsTableHead from './AuthorsTableHead';
 import AuthorsTableBody from './AuthorsTableBody';
+import { AuthorsTablePagination } from './AuthorsTablePagination';
+import { AuthorsListFilters } from './AuthorsListFilters';
 import { useStyles } from './styles/AuthorsList.styles';
+import i18n, { langTokens } from '../../../locales/localizationInit';
+import {
+  fetchExpertsAutorsList,
+  selectExpertsData,
+  selectExpertsMeta,
+  setExpertsStateToInit,
+  selectExpertsByIds,
+} from '../../../models/experts';
+import { useActions } from '../../../shared/hooks';
 
 export const AuthorsList: React.FC = () => {
   const classes = useStyles();
-  /* Temporary Mock for authors info until getting API */
+  const [boundFetchExpertsAutorsList, boundExpertsStateToInit] = useActions([
+    fetchExpertsAutorsList,
+    setExpertsStateToInit,
+  ]);
+
+  const { expertIds, totalPages, totalElements } = useSelector(
+    selectExpertsData,
+  );
+  const {
+    pageNumber,
+    size,
+    sort: { order, sortBy },
+    textFields: { author },
+  } = useSelector(selectExpertsMeta);
+  const experts = selectExpertsByIds(expertIds);
 
-  const authors: IAdminLabExpert[] = [
-    {
-      id: 10,
-      firstName: 'Марія',
-      lastName: 'Марієнко',
-      region: 'Київська область',
-      dateOfCreation: '12.03.2022',
-      dateOfEdition: '12.03.2022',
-      bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
-      mainInstitution: {
-        id: 5,
-        city: {
-          id: 55,
-          name: 'Бровари',
-        },
-        name: 'Medical Idea',
-      },
-    },
-    {
-      id: 6,
-      firstName: 'Палана',
-      lastName: 'Литвинова',
-      region: 'Дніпропетровська область',
-      dateOfCreation: '12.03.2022',
-      dateOfEdition: '12.03.2022',
-      bio:
-        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ',
-      mainInstitution: {
-        id: 4,
-        city: {
-          id: 119,
-          name: 'Дніпро',
-        },
-        name: 'Медікум',
-      },
-    },
-    {
-      id: 6,
-      firstName: 'Таржеман',
-      lastName: 'Соколов',
-      region: 'Дніпропетровська область',
-      dateOfCreation: '12.03.2022',
-      dateOfEdition: '12.03.2022',
-      bio:
-        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ',
+  const maxCouldBePerPage = (pageNumber + 1) * Number(size);
```

Comment:
sorry, I'm kind of lost here

may you describe what's logic behind `from` and `to` by few examples?

Reply:
![Screenshot 2022-10-13 at 12 43 37](https://user-images.githubusercontent.com/40029364/195563081-bb66f1c9-0a39-4d3e-a3de-4e11b0c9be1c.png)

Reply:
![Screenshot 2022-10-13 at 17 26 35](https://user-images.githubusercontent.com/40029364/195624314-c254d022-86b8-44da-8938-fb2dcadf33cd.png)

Reply:
I think it could be simplified by checking for `isLastPage`:

```
const rangeStart = pageNumber * pageSize;
const rangeEnd = isLastPage ? totalElements : (pageNumber + 1 ) * pageSize;
```

that's I've renamed `from` and `end` just for clarity; I'm not insistin on using that name